argumentsExt/Resolver/ArgumentsResolver.php

```diff
@@ -48,25 +46,27 @@ public function __construct(ScenarioStateInitializer $store, Reader $reader)
      */
     public function resolve(\ReflectionMethod $function, array $arguments)
     {
-        // No `@ScenarioStateArgument` annotation found
-        if (null === $this->reader->getMethodAnnotation($function, ScenarioStateArgument::class)) {
+        // No `@StepArgumentInjectorArgument` annotation found
+        if (null === $this->reader->getMethodAnnotation($function, StepInjectorArgument::class)) {
             return $arguments;
         }
 
         $paramsKeys = array_map(function (\ReflectionParameter $element) {
             return $element->getName();
         }, $function->getParameters());
-        $store = $this->store->getStore();
 
         // Prepare arguments from annotations
-        /** @var ScenarioStateArgument[] $annotations */
+        /** @var StepArgumentInjectorArgument[] $annotations */
         $annotations = $this->reader->getMethodAnnotations($function);
         foreach ($annotations as $annotation) {
-            if ($annotation instanceof ScenarioStateArgument &&
-                in_array($annotation->getArgument(), $paramsKeys) &&
-                $store->hasStateFragment($annotation->getName())
+            if ($annotation instanceof StepInjectorArgument &&
+                in_array($annotation->getArgument(), $paramsKeys)
             ) {
-                $arguments[$annotation->getArgument()] = $store->getStateFragment($annotation->getName());
+                foreach ($this->hookers as $hooker) {
+                    if ($hooker->hasStepArgumentFor($annotation->getName())) {
```

Comment:
this `if` block is business logic and should be moved elsewhere in businessland.